### PR TITLE
Diktat CLI runner

### LINKED ISSUE #1557: Add main class for diktat.jar
~~Produce additional artifact with required ktlint version.
It can be zip\tar.gz\tgz archive with final diktat fat jar and ktlint with proper version~~

Add main class for diktat which runs `KtLint.format` or `KtLint.lint`.
Need to support the same arguments as `Benedikt`

* [x] Add the main class (#1566 hasn't been cherry-picked into `master`).
* [x] `Main-Class` attribute in `META-INF/MANIFEST.MF` of the fat jar
* [x] ~~`Class-Path` attribute in `META-INF/MANIFEST.MF` of the fat jar~~

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
* added `diktat-cli` to run diktat as a cli command

Cherry-picked https://github.com/saveourtool/diktat/pull/1566

It closes #1557

### CODE REVIEW

File: .github/workflows/build_and_test.yml [hunk 1]
```diff
@@ -77,37 +84,55 @@ jobs:
...
+      - name: Download diktat cli jar
+        if: matrix.type == 'cli'
+        uses: actions/download-artifact@v3
+        with:
+          name: diktat-cli
+
+      - name: Generate run command using cli
+        if: matrix.type == 'cli'
+        run: |
+          filename=$(ls diktat-cli-*.jar)
```
Comment (0x6675636b796f75676974687562) [inline .github/workflows/build_and_test.yml:128]:
```suggestion
          filename=$(ls -1 diktat-cli-*.jar | head -n1)
```

File: .github/workflows/build_and_test.yml [hunk 2]
```diff
@@ -77,37 +84,55 @@ jobs:
...
+        if: matrix.type == 'ktlint'
         uses: actions/download-artifact@v3
         with:
           name: diktat-ruleset
 
-      - name: Retrieve diktat jar file name
+      - name: Generate run command using ktlint
+        if: matrix.type == 'ktlint'
         run: |
           filename=$(ls diktat-*.jar)
```
Comment (0x6675636b796f75676974687562) [inline .github/workflows/build_and_test.yml:115]:
```suggestion
          filename=$(ls -1 diktat-*.jar | head -n1)
```

File: .github/workflows/build_and_test.yml [hunk 3]
```diff
@@ -77,37 +84,55 @@ jobs:
...
+        if: matrix.type == 'cli'
+        uses: actions/download-artifact@v3
+        with:
+          name: diktat-cli
+
+      - name: Generate run command using cli
+        if: matrix.type == 'cli'
+        run: |
+          filename=$(ls diktat-cli-*.jar)
+          echo DIKTAT_RUN="java -jar $filename" >> $GITHUB_ENV
```
Comment (0x6675636b796f75676974687562) [inline .github/workflows/build_and_test.yml:129]:
```suggestion
          echo DIKTAT_RUN="java -jar \"$filename\"" >> $GITHUB_ENV
```

File: .github/workflows/build_and_test.yml [hunk 4]
```diff
@@ -121,14 +146,14 @@ jobs:
...
-          java -jar ktlint -R ${{ env.DIKTAT_JAR }} --disabled_rules=standard,experimental,test,custom "$PWD/examples/maven/src/main/kotlin/Test.kt" &>out.txt
+          ${{ env.DIKTAT_RUN }} "$PWD/examples/maven/src/main/kotlin/Test.kt" &>out.txt
         shell: bash
 
       - name: Run diKTat from cli on windows (absolute paths)
         continue-on-error: true
         if: runner.os == 'Windows'
         run: |
-          java -jar ktlint -R ${{ env.DIKTAT_JAR }} --disabled_rules=standard,experimental,test,custom "%cd%/examples/maven/src/main/kotlin/Test.kt" > out.txt 2>&1
+          ${{ env.DIKTAT_RUN }} "%cd%/examples/maven/src/main/kotlin/Test.kt" > out.txt 2>&1
```
Comment (0x6675636b796f75676974687562) [inline .github/workflows/build_and_test.yml:156]:
```suggestion
          ${{ env.DIKTAT_RUN }} "%cd%/examples/maven/src/main/kotlin/Test.kt" &>out.txt
```

File: diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt [hunk 5]
```diff
@@ -0,0 +1,242 @@
...
+    /**
+     * Configure logger level using [logLevel]
+     */
+    fun configureLogger() {
+        // set log level
+        LogManager.getContext(false)
+            .let { it as LoggerContext }
+            .also { ctx ->
+                ctx.configuration.rootLogger.level = when (logLevel) {
+                    Level.ERROR -> org.apache.logging.log4j.Level.ERROR
```
Comment (0x6675636b796f75676974687562) [inline diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt:53]: Up to you, but you can 

```kotlin
import org.apache.logging.log4j.Level as Log4jLevel
```

or even 

```kotlin
import org.apache.logging.log4j.Level.ERROR as LOG4J_ERROR
```

at the beginning of your file.

Feel free to leave w/o resolution, though.

File: diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt [hunk 6]
```diff
@@ -0,0 +1,242 @@
...
+    )
+
+    private fun getFiles(sourceRootDir: Path): Collection<Path> = patterns
+        .asSequence()
+        .flatMap { pattern ->
+            pattern.tryToPathIfExists()?.let { sequenceOf(it) }
+                ?: sourceRootDir.walkByGlob(pattern)
+        }
+        .filter { file -> file.isKotlinCodeOrScript() }
+        .distinct()
```
Comment (0x6675636b796f75676974687562) [inline diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt:90]: I'd put `distinct()` _after_ `toAbsolutePath()` and `normalize()` are called.

File: diktat-cli/src/test/kotlin/org/cqfn/diktat/util/CliUtilsKtTest.kt [hunk 7]
```diff
@@ -0,0 +1,92 @@
+package org.cqfn.diktat.util
+
+import org.jetbrains.kotlin.konan.file.File
```
Comment (0x6675636b796f75676974687562) [inline diktat-cli/src/test/kotlin/org/cqfn/diktat/util/CliUtilsKtTest.kt:3]: Are you sure this is not `java.io.File`?

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt (hunk 5 above)
Comment (nulls) [reply diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt:53]: it's used at in a single place. will leave as is

File: diktat-cli/src/test/kotlin/org/cqfn/diktat/util/CliUtilsKtTest.kt (hunk 7 above)
Comment (nulls) [reply diktat-cli/src/test/kotlin/org/cqfn/diktat/util/CliUtilsKtTest.kt:3]: yeap, chnaged

File: diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt (hunk 6 above)
Comment (nulls) [reply diktat-cli/src/main/kotlin/org/cqfn/diktat/cli/DiktatProperties.kt:90]: done

File: .github/workflows/build_and_test.yml (hunk 4 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:156]: updated

File: .github/workflows/build_and_test.yml (hunk 2 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:115]: fixed

File: .github/workflows/build_and_test.yml (hunk 1 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml]: done

File: .github/workflows/build_and_test.yml (hunk 4 above)
Comment (nulls) [reply .github/workflows/build_and_test.yml:157]: need to revert, since it's running on Windows and Windows doesn't support such construction

**Pull request merged**